src/Management/src/Endpoint/Services/ServiceDescriptorConverter.cs

```diff
@@ -0,0 +1,61 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileSystemGlobbing.Internal;
+using Steeltoe.Management.Endpoint.Hypermedia;
+
+namespace Steeltoe.Management.Endpoint.Services;
+internal class ServiceDescriptorConverter : JsonConverter<ServiceContextDescriptor>
+{
+    public override ServiceContextDescriptor Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        throw new NotImplementedException();
+    }
+
+    public override void Write(Utf8JsonWriter writer, ServiceContextDescriptor value, JsonSerializerOptions options)
+    {
+        writer.WriteStartObject();
+        writer.WritePropertyName("contexts");
+        writer.WriteStartObject();
+     
+        foreach (var contextKey in value.Keys)
+        {
+           
+            writer.WritePropertyName(contextKey);
+            writer.WriteStartObject();
+            writer.WritePropertyName("beans");
+            writer.WriteStartObject();
+            foreach (var (key, serviceDescriptor) in value[contextKey])
+            {
+                var name = serviceDescriptor.ServiceType.Name;
+             
+                writer.WritePropertyName(name);
+                writer.WriteStartObject();
+                writer.WriteString("scope", serviceDescriptor.Lifetime.ToString());
+                writer.WriteString("type", key);
+                writer.WriteString("resource", serviceDescriptor.ServiceType.AssemblyQualifiedName);
+                writer.WritePropertyName("dependencies");
+                
+                writer.WriteStartArray();
+                if (serviceDescriptor.ImplementationType != null)
+                {
+                    foreach (var constructor in serviceDescriptor.ImplementationType.GetConstructors())
+                    {
+                       
+                        writer.WriteStringValue(constructor.ToString());
```

Comment:
Technically yes, but from a practical standpoint, the constructors are the various sources of dependencies provided by the service container and it results in the same information and I tested that it is readable in at least SBA.

Reply:
I could not find an easy way to ask BCL for this and as you said implementing ourselves is complicated. So what I did here is to return a list of dependencies from all constructors.

Reply:
This is better, but it prints the parameters, not their types. The names are just noise (and Spring doesn't do that). Also, there's no `?` suffix if the parameter has a default value.

Please use the following instead:
```c#
private static string GetParameterType(ParameterInfo parameterInfo)
{
    return parameterInfo.HasDefaultValue ? $"{parameterInfo.ParameterType}?" : parameterInfo.ParameterType.ToString();
}
```

Also, using a `List<string>` for the merge is inefficient, use a `HashSet<String>` instead (and remove the existence check).

Reply:
I'm seeing the following behavior in .NET:
1. A constructor that has an optional parameter _without_ a default value is _discarded_ when the dependency _cannot_ be resolved.
2. A constructor that has an optional parameter _without_ a default value is _used_ when the dependency _can_ be resolved.
3. A constructor that has an optional parameter _with_ a default value is _used_, even when the dependency _cannot_ be resolved.

This leads me to believe we should emit the `?` suffix when the parameter is optional, regardless of whether it has a default value.

Reply:
Consider the next case:
```c#
#nullable enable

public sealed class ExampleService : IExampleService
{
    public ExampleService(Dependency dependency, int some = 3)
    {
    }

    public ExampleService(Dependency? dependency = null, bool other = true)
    {
    }
}

public class Dependency
{
}
```
which would print:
```json
"dependencies": [
  "Steeltoe.Actuators.ExampleServices.Dependency?",
  "Boolean?",
  "Steeltoe.Actuators.ExampleServices.Dependency",
  "Int32?"
]
```
Because of the parameter collapsing, the type now occurs twice. It's not obvious what belongs where. Perhaps we should refrain from adding the `?` suffix at all.

Reply:
As explained above, adding the `?` suffix is sometimes helpful and confusing at other times. So lets' not add the suffix, for now.